tests/basic/storage_pool_test.go

```diff
@@ -6302,20 +6328,787 @@ var _ = Describe("{PoolResizeInvalidPoolID}", func() {
+
+		// Get Node Details of the Pool with IO
+		nodeDetail, err := GetNodeWithGivenPoolID(poolUUID)
+		log.FailOnError(err, "Failed to get Node Details from PoolUUID [%v]", poolUUID)
+		log.InfoD("Pool with UUID [%v] present in Node [%v]", poolUUID, nodeDetail.Name)
+
+		poolToBeResized, err := GetStoragePoolByUUID(poolUUID)
+		log.FailOnError(err, "error getting drive size for pool [%s]", poolToBeResized.Uuid)
+
+		allPoolsOnNode, err := GetPoolsDetailsOnNode(*nodeDetail)
+		log.FailOnError(err, fmt.Sprintf("Failed to get all Pools present in Node [%s]", nodeDetail.Name))
+
+		drvSize, err := getPoolDiskSize(poolToBeResized)
+		log.FailOnError(err, fmt.Sprintf("Failed to get pool using UUID [%s]", poolUUID))
+
+		expectedSize := (poolToBeResized.TotalSize / units.GiB) + drvSize
+		expectedSize = roundUpValue(expectedSize)
+		expectedSizeWithJournal := expectedSize
+
+		isjournal, err := isJournalEnabled()
+		log.FailOnError(err, "Failed to check is journal enabled")
+
+		if isjournal {
+			expectedSizeWithJournal = expectedSizeWithJournal - 3
+		}
+		log.InfoD("Current Size of the pool %s is %d",
+			poolToBeResized.Uuid,
+			poolToBeResized.TotalSize/units.GiB)
+
+		err = Inst().V.ExpandPool(poolToBeResized.Uuid,
+			api.SdkStoragePool_RESIZE_TYPE_AUTO,
+			expectedSize)
+		dash.VerifyFatal(err,
+			nil,
+			"Pool expansion init successful?")
+
+		resizeErr := waitForPoolToBeResized(expectedSize, poolToBeResized.Uuid, isjournal)
+		dash.VerifyFatal(resizeErr, nil,
+			fmt.Sprintf("Verify pool [%s] on expansion using auto option", poolToBeResized.Uuid))
+
+		// Restart Px and wait for Driver to come up after pool expanded
+		restartPx := func() error {
+			stepLog = fmt.Sprintf("Restart PX on node %s", nodeDetail.Name)
+			err := Inst().V.RestartDriver(*nodeDetail, nil)
+			if err != nil {
+				return err
+			}
+			err = Inst().V.WaitDriverUpOnNode(*nodeDetail, addDriveUpTimeOut)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		// Restarting Px after expanding the pool
+		log.FailOnError(restartPx(), fmt.Sprintf("Failed to restart Px on the Node [%v]", nodeDetail.Name))
+
+		// Expand Pool using Add Drive and verify if the Pool is expanded successfully
+		err = Inst().V.ExpandPool(poolToBeResized.Uuid,
+			api.SdkStoragePool_RESIZE_TYPE_ADD_DISK,
+			expectedSize)
+		dash.VerifyFatal(err,
+			nil,
+			"Pool expansion init successful?")
+
+		resizeErr = waitForPoolToBeResized(expectedSize, poolUUID, isjournal)
+		dash.VerifyFatal(resizeErr, nil,
+			fmt.Sprintf("Verify pool [%s] on expansion using auto option", poolUUID))
+
+		// Restarting Px after pool is resizied
+		log.FailOnError(restartPx(), fmt.Sprintf("Failed to restart Px on the Node [%v]", nodeDetail.Name))
+
+		allPoolsOnNode_afterResize, err := GetPoolsDetailsOnNode(*nodeDetail)
+		log.FailOnError(err, fmt.Sprintf("Failed to get all Pools present in Node [%s]", nodeDetail.Name))
+		dash.VerifyFatal(len(allPoolsOnNode) <= len(allPoolsOnNode_afterResize), true,
```

Comment:
here expansions are happening using pool expand rt? If so I dont think it will create any new pools

Reply:
yes, here we are failing if the new pool is created ,

Reply:
this scenario was covered beacuse of PWX-27163